core/shared/src/main/scala/scalaz/zio/Queue.scala

```diff
@@ -32,12 +32,12 @@ class Queue[A] private (capacity: Int, ref: Ref[State[A]]) {
    * the queue.
    */
   final def offer[E](a: A): IO[E, Unit] = {
-    val acquire: (Promise[E, Unit], State[A]) => (IO[Void, Boolean], State[A]) = {
+    val acquire: (Promise[E, Unit], State[A]) => (IO[Nothing, Boolean], State[A]) = {
```

Comment:
Can we change the error type of Promise to Nothing?
Because `Promise.bracket` should return an `IO[Nothing, B]` too, which is the IO of the promise(in `acquire` and it would be the same in `release`).

Reply:
`offer` cannot fail, so we can change this to: `def offer(a: A): IO[Nothing, Unit]`.

Look to see if we `error` any promise. I don't think we do. So we can change `E` in `Promise` to `Nothing`, too.

Reply:
Ok, thank you :)